src/core/qgsexpressioncontext.cpp

```diff
@@ -778,11 +778,15 @@ QgsExpressionContextScope *QgsExpressionContextUtils::projectScope( const QgsPro
     scope->setVariable( it.key(), it.value(), true );
   }
 
+  QString projectPath = project->absoluteFilePath();
+  QString projectFolder = QFileInfo( projectPath ).path();
+  QString projectFilename = QFileInfo( projectPath ).fileName();
```

Comment:
Should this be the project baseName instead? (So it has a value for postgres projects)

Reply:
Yeah, can we do that then?

What do you think about the project_path variable too? It would be nice to have some way to insert the postgres equivalent in layout footers.

Reply:
Cool - I will add "project_basename" that would work for non-file based projects, and I will update "project_path" to use project's URI for non-file based projects.